### ISSUE TITLE
Add proxy support to Try It

### ISSUE THREAD

**Issue by marcelltoth** [stoplightio/elements#1037]
## Overview

When using TryIt we should support interacting with APIs that have not enabled CORS headers for whatever domain the elements based documentation is running on.

- elements users can provide their own CORS proxy as a property, otherwise its all done directly.
- elements-dev-portal can also provide their own, but by default will use the hosted CORS proxy we set up for the demo at `https://cors.stoplight.io/`.

## Requirements

Let's add a `proxy` prop to both public components, which takes a string. As we are using `fetch()` try and find some portable solution that works in the browser. node-fetch for instance can do this:

```
const fetch = require('node-fetch');
const HttpsProxyAgent = require('https-proxy-agent');

(async () => {
    const proxyAgent = new HttpsProxyAgent('http://46.250.171.31:8080');
    const response = await fetch('https://httpbin.org/ip?json', { agent: proxyAgent});
    const body = await response.text();
    console.log(body);
})();
```

If that is not going to work, prepend it to the URL as we do for the demo:

```
const url = `https://cors.stoplight.io/${actualUrl}`;
```

## Example

```
<API apiDescriptionUrl="https://some-domain/oas.yaml" proxy="https://cors-anywhere.example.io/" />` 
```

## Out of Scope

This will eventually be replaced with hosted Prism Proxy, but that will require some work to do right. Let's unblock people with this existing CORS proxy then move over to Prism Proxy for elements-dev-portal when we can.

**Comment by marcelltoth** [comment]
@marbemac Should this (and the followup) be in the Early access release or is it a nice to have?
